src/Raven.Client/Documents/Session/AsyncSessionDocumentTimeSeries.cs

```diff
@@ -220,16 +260,51 @@ await ServeFromCacheOrGetMissingPartsFromServerAndMerge(from ?? DateTime.MinValu
                     From = ranges[ranges.Count - 1].To,
                     To = to
                 });
+
+                for (int i = fromRangeIndex < 0 ? 0 : fromRangeIndex; i < toRangeIndex; i++)
+                {
+                    count += ranges[i].Entries.Length;
+                }
             }
 
+            var partialPageSize = pageSize != int.MaxValue? pageSize - count + 1 : pageSize;
             // get all the missing parts from server
 
             Session.IncrementRequestCount();
 
             var details = await Session.Operations.SendAsync(
-                    new GetMultipleTimeSeriesOperation(DocId, rangesToGetFromServer, start, pageSize), Session._sessionInfo, token: token)
+                    new GetMultipleTimeSeriesOperation(DocId, rangesToGetFromServer, start, partialPageSize), Session._sessionInfo, token: token)
                 .ConfigureAwait(false);
 
+            var changeStart = true;
+            foreach (var tsrr in details.Values.SelectMany(detail => detail.Value))
+            {
+                if (tsrr.Entries.Length == 0) continue;
+
+                if (changeStart)
+                {
+                    tsrr.From = tsrr.Entries[0].Timestamp;
+                    changeStart = false;
+                }
+
+                if (partialPageSize != tsrr.Entries.Length) continue;
+
+                Session.IncrementRequestCount();
+
+                var temp = await Session.Operations.SendAsync(
```

Comment:
1. Can't the server handle this case and return the wanted results ?
2. If this call is indeed necessary, we need to collect all ranges and use `GetMultipleTimeSeriesOperation`.

Reply:
I will move everything to the server